src/main/java/com/google/step/snippet/external/StackOverflowClient.java

```diff
@@ -173,7 +174,11 @@ private String getResponse(String url, String fieldParam) {
       return null;
     }
     JSONObject json = new JSONObject(responseBody.toString());
-    String res = json.getJSONArray(ITEM_PARAMETER).getJSONObject(0).get(fieldParam).toString();
+    JSONArray items = json.getJSONArray(ITEM_PARAMETER);
+    if (items == null || items.length() == 0) {
+      return null;
+    }
+    String res = items.getJSONObject(0).get(fieldParam).toString();
     if (response.getStatusLine().getStatusCode() != 200) {
```

Comment:
Can follow a similar style in SearchServlet #18 :

```
if (json.has(ITEM_PARAMTER) && json.getAsJsonArray(ITEM_PARAMTER).size() > 0) {
    JsonObject item = obj.getAsJsonArray(ITEM_PARAMTER).get(0).getAsJsonObject();
    if (item.has(fieldParam)) {
        String res = item.get(fieldParam).getAsString();
    }
}
```

Reply:
both ways look fine i think..?

Reply:
Uh, defer to Andrew and Joan?

But I still don't think you're safely accessing the fieldParam here?